catalyst/core/callbacks/optimizer.py

```diff
@@ -88,6 +100,16 @@ def on_stage_start(self, runner: IRunner) -> None:
         self._optimizer = runner.get_attr(
             key="optimizer", inner_key=self.optimizer_key
         )
+        if runner.device.type == "xla":
+            if self.use_xla_barrier:
+                self._optimizer_step_fn = lambda optimizer: optimizer_step(
+                    optimizer, barrier=True
+                )
+            else:
+                self._optimizer_step_fn = optimizer_step
+        else:
+            self._optimizer_step_fn = lambda optimizer: optimizer.step()
```

Comment:
could we just specify 2 extra methods for callback?
```python
def _optimizer_step_gpu()...
  self._optimizer.step()

def _optimizer_step_tpu()
  ???
  
```

Reply:
just don't like the lambda usage :)

Reply:
sure, thanks